File: day07/Day07.scala
import scala.io.Source.stdin

@main def main(idx: Int) =
  val xs = stdin.getLines().next.split(',').map{_.toLong}
  val f : Long => Long = if idx == 1 then identity else (x => x * (x + 1) / 2)
  print((0L to xs.max).map{i => xs.map{j => f((i - j).abs)}.sum}.min)

File: day07/Day07.rs
use std::convert::identity;
use std::env;
use std::io;

fn main() {
    let nums: Vec<i64> = io::stdin()
        .lines()
        .next()
        .unwrap()
        .unwrap()
        .split(",")
        .map(|s| s.parse().unwrap())
        .collect();
    let top = *nums.iter().max().unwrap();

    let f: fn(i64) -> i64 = if env::args().nth(1).unwrap() == "1" {
        identity
    } else {
        |x| x * (x + 1) / 2
    };

    let result: i64 = (0..=top)
        .map(|i| nums.iter().map(|j| f((i - j).abs())).sum())
        .min()
        .unwrap();

    println!("{}", result);
}
